gadopt/gplates.py

```diff
@@ -0,0 +1,449 @@
+            >>> connector.get_plate_velocities(model_time=100)
+        """
+
+        # Rotation model(s)
+        self.rotation_model = pygplates.RotationModel(rotation_filenames)
+
+        # Topological plate polygon feature(s).
+        self.topology_features = []
+        for fname in topology_filenames:
+            for f in pygplates.FeatureCollection(fname):
+                self.topology_features.append(f)
+        # geologic_zero is the same as model_time=0
+        self.geologic_zero = geologic_zero
+
+        # time velocity interpolations
+        self.delta_time = delta_time
+
+        # seeds are equidistantial points generate on a sphere
+        self.seeds = self._fibonacci_sphere(samples=int(nseeds))
+        # number of neighbouring points that will be used for interpolation
+
+        self.nneighbours = nneighbours
+        self.velocity_domain_features = (
+            self._make_GPML_velocity_feature(
+                self.seeds
+            )
+        )
+
+        # last reconstruction time
+        self.reconstruction_time = None
+
+        # Flag to know when to recalculate surface velocities.
+        self.recalculation_flg = False
+
+        # Factor to scale plate velocities to RMS velocity of model,
+        # the definition: RMS_Earth / RMS_Model is used
+        # This is specially used in low-Rayleigh-number simulations
+        self.scaling_factor = scaling_factor
+
+    # setting the time that we are interested in
+    def get_plate_velocities(self, target_coords, model_time):
+        """
+        Returns plate velocities for the specified target coordinates at the top boundary of a sphere,
+        for a given model time, by integrating plate tectonic reconstructions from pyGplates.
+
+        This method calculates new plate velocities.
+        It utilizes the cKDTree data structure for efficient nearest neighbor
+        searches to interpolate velocities onto the mesh nodes.
+
+        Args:
+            target_coords (array-like): Coordinates of the points at the top of the sphere.
+            model_time (float): The model time for which plate velocities are to be calculated and assigned.
+                This time is converted to geological time and used to extract relevant plate motions
+                from the pyGplates model.
+
+        Raises:
+            Exception: If the requested reconstruction time is geologically negative, indicating an issue with
+                the time conversion.
+
+        Notes:
+            - The method uses conversions between non-dimensionalised time and geologic time.
+            - Velocities are non-dimensionalised and scaled for the simulation before being applied.
+        """
+
+        # Raising an error if the user is asking for invalid time
+        if self.ndtime2geotime(ndtime=model_time) < 0:
+            raise Exception(
+                ("pyGplates: geologic time is being negative!"
```

Comment:
Maybe a `ValueError` rather than a more generic `Exception`? Also, I don't think this error should be attributed to pygplates, since it's specific to this connector code.

Reply:
Agreed.